SparseGrids/tsgCoreOneDimensional.cpp

```diff
@@ -392,6 +393,168 @@ const char* OneDimensionalMeta::getHumanString(TypeOneDRule rule){
     }
 }
 
+// Generate the n roots of the n-th degree orthogonal polynomial.
+// ref_points and ref_points are used in the computation of the integrals that form the elements of the Jacobi matrix.
+double poly_eval(const std::vector<double> &roots, double x) {
+    double eval_value = 1.0;
+    for (size_t i=0; i<roots.size(); i++) {
+        eval_value *= (x - roots[i]);
+    }
+    return eval_value;
+}
+std::vector<std::vector<double>> OneDimensionalExoticQuad::getRootCache(
+        const int n,
+        std::function<double(double)> weight_fn,
+        const std::vector<double> &ref_weights,
+        const std::vector<double> &ref_points) {
+
+    // Initialize and compute the integral weights corresponding to weight_fn.
+    std::vector<std::vector<double>> root_cache(n);
```

Comment:
rename `roots_cache` to `roots` and remove `roots`

Reply:
Changed.